src/Tasks/Common/Resources/Strings.resx

```diff
@@ -937,5 +939,14 @@ You may need to build the project on another operating system or architecture, o
     <value>NETSDK1209: The current Visual Studio version does not support targeting {0} {1}.  Either target {0} {2} or lower, or use Visual Studio version {3} or higher</value>
     <comment>{StrBegin="NETSDK1209: "}</comment>
   </data>
-  <!-- The latest message added is UnsupportedVisualStudioVersion. Please update this value with each PR to catch parallel PRs both adding a new message -->
+  <data name="IsAotCompatibleUnsupported" xml:space="preserve">
+    <value>NETSDK1210: IsAotCompatible is not supported for the target framework. Consider multi-targeting to a supported framework to enable ahead-of-time compilation analysis, and set IsAotCompatible only for the supported frameworks. For example:
+&lt;IsAotCompatible Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))"&gt;true&lt;/IsAotCompatible&gt;</value>
+    <comment>{StrBegin="NETSDK1210: "}</comment>
+  </data>
+  <data name="EnableSingleFileAnalyzerUnsupported" xml:space="preserve">
+    <value>NETSDK1211: EnableSingleFileAnalyzer is not supported for the target framework.</value>
```

Comment:
Maybe also add the example with the conditional syntax - since we don't have an `IsSingleFileCompatible` the `EnableSingleFileAnalyzer` effectively acts like it - so the guidance should now apply to it.